stubs/openpyxl/openpyxl/cell/rich_text.pyi

```diff
@@ -0,0 +1,28 @@
+from collections.abc import Iterable
+from typing import Any, overload
+from typing_extensions import Self, TypeAlias
+
+from openpyxl.descriptors import Strict, String, Typed
+
+class TextBlock(Strict):
+    font: Typed
+    text: String
+
+    def __init__(self, font: Typed, text: String) -> None: ...
+    def __eq__(self, other: TextBlock) -> bool: ...  # type: ignore[override]
+
+# CellRichText accepts any object that can be converted to a string,
+# which is every object.
+_HasStr: TypeAlias = Any
+
+class CellRichText(list[_HasStr]):
+    @overload
+    def __init__(self, __args: list[_HasStr] | tuple[_HasStr]) -> None: ...
+    @overload
+    def __init__(self, *args: _HasStr) -> None: ...
+    @classmethod
+    def from_tree(cls, node) -> Self: ...
+    def __add__(self, arg: _HasStr) -> CellRichText: ...
+    def append(self, arg: _HasStr) -> None: ...
+    def extend(self, arg: Iterable[_HasStr]) -> None: ...
+    def as_list(self) -> list[str]: ...
```

Comment:
Sorry for going back and forth on this. I just took another look at the source code and, while the implementation is indeed pretty permissive, the docstring and comments are fairly clear that the members of the list should be either "simple strings or TextBlock() instances". So maybe we could be stricter in our annotations here:

```suggestion
class CellRichText(list[str | TextBlock]):
    @overload
    def __init__(self, __args: list[str] | list[TextBlock] | list[str | TextBlock] | tuple[str | TextBlock, ...]) -> None: ...
    @overload
    def __init__(self, *args: str | TextBlock) -> None: ...
    @classmethod
    def from_tree(cls, node) -> Self: ...
    def __add__(self, arg: Iterable[str | TextBlock]) -> CellRichText: ...
    def append(self, arg: str | TextBlock) -> None: ...
    def extend(self, arg: Iterable[str | TextBlock]) -> None: ...
    def as_list(self) -> list[str]: ...
```

Reply:
Sounds like a good idea.